cpp/examples/glct_secir.cpp

```diff
@@ -0,0 +1,171 @@
+        {50 * 0.25, 50 * 0.75},
+        {10 * 0.3, 10 * 0.3, 5 * 0.3, 3 * 0.3, 2 * 0.3, 10 * 0.7, 10 * 0.7, 5 * 0.7, 3 * 0.7, 2 * 0.7},
+        {20},
+        {10}};
+
+    // Assert that initial_populations has the right shape.
+    if (initial_populations.size() != (size_t)InfectionState::Count) {
+        mio::log_error("The number of vectors in initial_populations does not match the number of InfectionStates.");
+        return 1;
+    }
+    if ((initial_populations[(size_t)InfectionState::Susceptible].size() !=
+         LctState::get_num_subcompartments<InfectionState::Susceptible>()) ||
+        (initial_populations[(size_t)InfectionState::Exposed].size() != NumExposed) ||
+        (initial_populations[(size_t)InfectionState::InfectedNoSymptoms].size() != NumInfectedNoSymptoms) ||
+        (initial_populations[(size_t)InfectionState::InfectedSymptoms].size() != NumInfectedSymptoms) ||
+        (initial_populations[(size_t)InfectionState::InfectedSevere].size() != NumInfectedSevere) ||
+        (initial_populations[(size_t)InfectionState::InfectedCritical].size() != NumInfectedCritical) ||
+        (initial_populations[(size_t)InfectionState::Recovered].size() !=
+         LctState::get_num_subcompartments<InfectionState::Recovered>()) ||
+        (initial_populations[(size_t)InfectionState::Dead].size() !=
+         LctState::get_num_subcompartments<InfectionState::Dead>())) {
+        mio::log_error("The length of at least one vector in initial_populations does not match the related number of "
+                       "subcompartments.");
+        return 1;
+    }
+
+    // Transfer the initial values in initial_populations to the model.
+    std::vector<ScalarType> flat_initial_populations;
+    for (auto&& vec : initial_populations) {
+        flat_initial_populations.insert(flat_initial_populations.end(), vec.begin(), vec.end());
+    }
+    for (size_t i = 0; i < LctState::Count; i++) {
+        model.populations[mio::Index<LctState>(i)] = flat_initial_populations[i];
+    }
+
+    // Set Parameters.
+    // Exposed.
+    model.parameters.get<mio::glsecir::StartingProbabilitiesExposed>() =
+        mio::glsecir::StartingProbabilitiesExposed().get_default(
+            LctState::get_num_subcompartments<InfectionState::Exposed>());
+    model.parameters.get<mio::glsecir::TransitionMatrixExposedToInfectedNoSymptoms>() =
+        mio::glsecir::TransitionMatrixExposedToInfectedNoSymptoms().get_default(
+            LctState::get_num_subcompartments<InfectionState::Exposed>(), 3.2);
+    // InfectedNoSymptoms.
+    Eigen::VectorXd StartingProbabilitiesInfectedNoSymptoms =
+        Eigen::VectorXd::Zero(LctState::get_num_subcompartments<InfectionState::InfectedNoSymptoms>());
+    StartingProbabilitiesInfectedNoSymptoms[0]                                         = 1 - 0.09;
+    StartingProbabilitiesInfectedNoSymptoms[(Eigen::Index)(
+        LctState::get_num_subcompartments<InfectionState::InfectedNoSymptoms>() / 2.)] = 0.09;
```

Comment:
Here, you assume that there is the same number of subcompartments for those individuals that will at some point transition to InfectedSymptoms as for those that will transition to Recovered, right? Can you mention that somewhere?